modules/core/include/opencv2/core/hal/intrin.hpp

```diff
@@ -166,7 +166,7 @@ CV_INTRIN_DEF_TYPE_TRAITS(schar, schar, uchar, uchar, short, int, int);
 CV_INTRIN_DEF_TYPE_TRAITS(ushort, short, ushort, ushort, unsigned, uint64, unsigned);
 CV_INTRIN_DEF_TYPE_TRAITS(short, short, ushort, ushort, int, int64, int);
 #if CV_FP16_TYPE
-CV_INTRIN_DEF_TYPE_TRAITS(__fp16, short, ushort, __fp16, float, double, float);
+CV_INTRIN_DEF_TYPE_TRAITS(hfloat, short, ushort, hfloat, float, double, float);
```

Comment:
@opencv-alalek , I thought `hfloat` is our platform-dependent wrapper for FP16 data? It would become `__fp16` on ARM and `_Float16` on x86_64 and RISC-V. Intrinsics will use own type internally, but they still need `v_load/v_store` interfaces with `hfloat`. Or did I misunderstand your comment?

Reply:
Discussed offline.
It seems fine to use `hfloat` as **external** interface for scalar/pointer FP16 types of universal SIMD intrinsics.